integration/tests/bft/passthrough/suite.go

```diff
@@ -0,0 +1,186 @@
+package passthrough
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/ghost/client"
+	"github.com/onflow/flow-go/insecure/attacknetwork"
+	"github.com/onflow/flow-go/integration/testnet"
+	"github.com/onflow/flow-go/integration/tests/lib"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// Suite represents a test suite evaluating the integration of the testnet against
+// happy path of Corrupted Conduit Framework (CCF) for BFT testing.
+type Suite struct {
+	suite.Suite
+	log                     zerolog.Logger
+	lib.TestnetStateTracker                      // used to track messages over testnet
+	cancel                  context.CancelFunc   // used to tear down the testnet
+	net                     *testnet.FlowNetwork // used to keep an instance of testnet
+	nodeConfigs             []testnet.NodeConfig // used to keep configuration of nodes in testnet
+	nodeIDs                 []flow.Identifier    // used to keep identifier of nodes in testnet
+	ghostID                 flow.Identifier      // represents id of ghost node
+	exe1ID                  flow.Identifier      // corrupted execution node 1
+	exe2ID                  flow.Identifier      // corrupted execution node 2
+	verID                   flow.Identifier      // corrupted verification node
+	PreferredUnicasts       string               // preferred unicast protocols between execution and verification nodes.
+	Orchestrator            *dummyOrchestrator
+	attackNet               *attacknetwork.AttackNetwork
+}
+
+// Ghost returns a client to interact with the Ghost node on testnet.
+func (s *Suite) Ghost() *client.GhostClient {
+	ghost := s.net.ContainerByID(s.ghostID)
+	cli, err := lib.GetGhostClient(ghost)
+	require.NoError(s.T(), err, "could not get ghost client")
+	return cli
+}
+
+// AccessClient returns a client to interact with the access node api on testnet.
+func (s *Suite) AccessClient() *testnet.Client {
+	chain := s.net.Root().Header.ChainID.Chain()
+	cli, err := testnet.NewClient(fmt.Sprintf(":%s", s.net.AccessPorts[testnet.AccessNodeAPIPort]), chain)
+	require.NoError(s.T(), err, "could not get access client")
+	return cli
+}
+
+// SetupSuite runs a bare minimum Flow network to function correctly with the following roles:
+// - Two collector nodes
+// - Four consensus nodes
+// - two corrupted execution node
+// - One corrupted verification node
+// - One ghost node (as an execution node)
+func (s *Suite) SetupSuite() {
+	logger := unittest.LoggerWithLevel(zerolog.InfoLevel).With().
+		Str("testfile", "suite.go").
+		Str("testcase", s.T().Name()).
+		Logger()
+	s.log = logger
+
+	blockRateFlag := "--block-rate-delay=1ms"
```

Comment:
Why do have this flag set to `1ms` for this test, but not other integration tests?

Reply:
Fixed in https://github.com/onflow/flow-go/pull/2397/commits/70dfcdae85d46c2a6da1e8f29e0b17b1edb7e732